### PR TITLE
⚡️ 🚑 App Service Hotfix 2.0

### LINKED ISSUE #45: Env locking breaking app service still
The environment locking allows the web app to be deployed, however it does not run successfully and will return 404 responses to curl requests as specified in the README. The locking script needs to be updated in order to place the pip requirements into the `requirements.txt` file if the Inference service is to run successfully.

### PR DESCRIPTION
**Pull request by peterhessey**
Closes #45.

[The previous Hotfix ](https://github.com/microsoft/InnerEye-Inference/pull/44) only addressed the surface-level issue of the `az webapp up` command not working. I naively did not properly manually test the deployed app, and turns out it was not running correctly at all: Azure App Services build their environments from the `requirements.txt` file (unsurprisingly) and as such the pip requirements _must_ be placed into that file.

This PR includes changes to the environment locking script to seperate the conda and pip package versions between `environment.yml` and `requirements.txt` respectively, as well as pinning `msrest==0.6.21` to circumvent certain incompatibilities as discussed in [this thread](https://github.com/Azure/azure-sdk-for-python/issues/24765).

### CODE REVIEW

**Review by ant0nsc — APPROVED** [review]

File: create_and_lock_environment.sh [hunk 1]
```diff
@@ -15,25 +15,35 @@ env_name="${name_arr[1]}"
 
 # clear old conda envs, create new one
 export CONDA_ALWAYS_YES="true"
-conda env remove --name ${env_name::-1}
```
[the remark this answers is not in the document]
Comment (fepegar) [reply create_and_lock_environment.sh:18]: Maybe there was a trailing whitespace after the env name in the YAML?

**Review by fepegar — APPROVED** [review]
Out of curiosity, what made you choose Bash for this script, instead of e.g. Python?

File: create_and_lock_environment.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterhessey) [reply create_and_lock_environment.sh:18]: This would make a lot of sense!

[the remark this answers is not in the document]
Comment (fepegar) [reply create_and_lock_environment.sh:18]: (the bot would have removed trailing whitespaces after committing and this would've never happened)

**Pull request merged**